activesupport/lib/active_support/inflector/transliterate.rb

```diff
@@ -75,13 +75,21 @@ def parameterize(string, sep = '-')
       # Turn unwanted chars into the separator
       parameterized_string.gsub!(/[^a-z0-9\-_]+/i, sep)
       unless sep.nil? || sep.empty?
-        re_sep = Regexp.escape(sep)
+        if sep == "-".freeze
```

Comment:
The comparison was the main thing I thought about.
If you call it without the second parameter it will have a different string "-" created but the comparison will work.